src/main.rs

```diff
@@ -18,66 +13,159 @@ mod util;
-            }
+        Some(("add", sub_matches)) => {
+            let t: Task = Task::from(
+                sub_matches
+                    .get_one::<String>("CONTENT")
+                    .expect("Failure in parsing task")
+                    .to_string(),
+            );
+
+            cli_add_task(&db, t);
         }
-        ArgumentType::Clear => {
-            if cli_clear(&db) {
-                println!("removed all tasks from database");
-            } else {
-                println!("couldn't remove all tasks from the database");
-            }
+        Some(("delete", sub_matches)) => {
+            let id = sub_matches
+                .get_one::<String>("ID")
+                .expect("Couldn't get id from input")
+                .to_string();
+
+            let feedback = cli_del_task(&db, id);
+            println!(
+                "{}",
+                if feedback {
+                    "deleted task"
+                } else {
+                    "failed to delete given task"
+                }
+            )
         }
-        ArgumentType::Export {
-            export_type,
-            file_name,
-        } => {
-            let data = cli_export(&db, export_type);
+        Some(("finish", sub_matches)) => {
+            let id = sub_matches
+                .get_one::<String>("ID")
+                .expect("Couldn't get id from input")
+                .to_string();
 
-            let file_name_with_extension = format!("{}.{}", file_name, export_type);
-            let mut file =
-                std::fs::File::create(file_name_with_extension).expect("Unable to open file");
-            write!(file, "{}", data).expect("Unable to write");
+            let feedback = cli_fin_task(&db, id);
+            println!(
+                "{}",
+                if feedback {
+                    "marked task as finished"
+                } else {
+                    "failed to mark task as finished"
+                }
+            )
         }
-        ArgumentType::Delete => {
-            if cli_del_task(&db, result.input.query.unwrap()) {
-                println!("deleted task");
-            } else {
-                println!("couldn't delete task");
-            }
+        Some(("clear", _)) => {
+            let feedback = cli_clear(&db);
+            println!(
+                "{}",
+                if feedback {
+                    "cleared database"
+                } else {
+                    "failed to clear database"
+                }
+            );
+        }
+        Some(("export", sub_matches)) => {
+            let temp = sub_matches
+                .get_one::<String>("fileformat")
+                .expect("Bad Fileformat");
+
+            let export_type = match temp.as_str() {
+                "csv" => ExportType::Csv,
+                "json" => ExportType::Json,
+                "tsv" => ExportType::Tsv,
+                _ => panic!("Unknown exporttype"),
+            };
```

Comment:
Since you moved the task parsing to `Task::from` with this pull request I encourage you to do the same for the `ExportType` enum. The conversion to a string is already there using the `Display` trait and imho the conversion in the other direction should as well be implemented by the enum to have both conversions at the same place.